pkg/codegen/go/gen.go

```diff
@@ -2545,6 +2545,10 @@ func (pkg *pkgContext) genResourceModule(w io.Writer) {
 		fmt.Fprintf(w, "func (m *module) Construct(ctx *pulumi.Context, name, typ, urn string) (r pulumi.Resource, err error) {\n")
 		fmt.Fprintf(w, "\tswitch typ {\n")
 		for _, r := range pkg.resources {
+			if r.IsOverlay {
```

Comment:
Do we need to add a test schema(s) for ensuring we don't leak overlays by mistake here and in other languages' codegen?

Reply:
Yeah, probably so.

Reply:
Done, and caught a bug in the process!